cidc_api/shared/gcloud_client.py

```diff
@@ -172,13 +172,51 @@ def revoke_download_access(user_email: str, trial_id: str, upload_type: str):
     policy = bucket.get_iam_policy(requested_policy_version=3)
     policy.version = 3
 
-    # find the index of the binding for this user/trial/upload_type combo
-    clean_bindings = []
-    for binding in policy.bindings:
-        if not _is_matching_conditional_binding(binding, member_id, prefix_expression):
-            clean_bindings.append(binding)
+    # find the policy binding for this user if one exists
+    binding_index, binding = _get_user_download_binding(policy, user_email)
+
+    # remove the relevant clause from the condition if it exists
+    if binding_index is not None:
+        expression = binding["condition"]["expression"]
+        if prefix_expression in expression:
```

Comment:
isn't that like an integrity error if prefix_expression *NOT* in expression? 
I suggest adding warning logging or something

Reply:
Sure, a warning sounds good.